google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +140,31 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    // ensures finite iteration
+    int maxDepth = 10;
+    while (throwable != null && maxDepth-- > 0) {
+      Throwable cause = throwable.getCause();
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || cause instanceof SSLException
+          || cause instanceof EOFException
+          || cause instanceof SocketException
+          || cause instanceof SocketTimeoutException) {
+        return true;
+      }
+      throwable = cause;
+    }
+    return false;
+  }
+
   private void sleepForAttempt(int attempt) {
+    long firstdelay = 1000;
+    long maxDelay = 120_000;
+    // exponential backoff
+    long delay = firstdelay * Math.min(1 << attempt, maxDelay);
```

Comment:
Also please add a comment about why an aggressive factor of 2 is appropriate here.